framework/src/fvkernels/FVCoupledPowerLawForce.C

```diff
@@ -0,0 +1,59 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVCoupledPowerLawForce.h"
+
+registerMooseObject("MooseApp", FVCoupledPowerLawForce);
+
+InputParameters
+FVCoupledPowerLawForce::validParams()
+{
+  InputParameters params = FVElementalKernel::validParams();
+
+  params.addClassDescription("Implements a source term proportional to the sum of "
+                             "up to three coupled variables, each raised to a given power "
+                             "law coefficent.");
+  params.addRequiredCoupledVar("u", "The coupled variable which provides the force");
+  params.addCoupledVar(
+      "v", "The second component of a vector coupled variable which provides the force");
+  params.addCoupledVar("w",
+                       "The third component of a vector coupled variable which provides the force");
+  params.addParam<Real>("coef", 1., "Coefficent multiplier for the coupled force term.");
+  params.addParam<MaterialPropertyName>(
+      "coef_name", 1., "Coefficent multiplier as a material property for the coupled force term.");
+  params.addParam<Real>("power_law_coef", 1., "Power law coefficent for the coupled force term.");
+
+  return params;
+}
+
+FVCoupledPowerLawForce::FVCoupledPowerLawForce(const InputParameters & parameters)
+  : FVElementalKernel(parameters),
+    _u(adCoupledValue("u")),
+    _v(adCoupledValue("v")),
+    _w(adCoupledValue("w")),
+    _coef(getParam<Real>("coef")),
+    _coef_mat_prop(getADMaterialProperty<Real>("coef_name")),
+    _power(getParam<Real>("power_law_coef"))
+{
+  // Check that at most one friction coefficient has been provided
+  if (parameters.isParamSetByUser("coef") && parameters.isParamSetByUser("coef_name"))
+    mooseError("FVCoupledPowerLawForce should be provided either a coefficient "
+               "or a coefficient material property");
+}
+
+ADReal
+FVCoupledPowerLawForce::computeQpResidual()
+{
+  if (!parameters().isParamSetByUser("coef_name"))
+    return -_coef *
+           (std::pow(_u[_qp], _power) + std::pow(_v[_qp], _power) + std::pow(_w[_qp], _power));
+  else
+    return -_coef_mat_prop[_qp] *
+           (std::pow(_u[_qp], _power) + std::pow(_v[_qp], _power) + std::pow(_w[_qp], _power));
```

Comment:
We should probably add an offset in here to prevent the AD error when `_power < 1` and a value is zero.